vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/scatter_plotly.tsx

```diff
@@ -0,0 +1,152 @@
+// Input component that takes nested object
+//   and shows the keys one level at a time.
+// Returns the last "Leaf" that the user selects.
+import React, {useState, useEffect, useMemo, useCallback} from 'react';
+import * as _ from 'lodash';
+
+import DropdownAutocomplete from 'etna-js/components/inputs/dropdown_autocomplete';
+import {InputBackendComponent} from './input_types';
+import TextInput from 'etna-js/components/inputs/text_input';
+import { JsxElement } from 'typescript';
+
+/*
+This input is closely tied to archimedes/functions/plotting/scatter_plotly.
+
+Major design notes:
+- It will have a set of advanced options that are shown/hidden via a toggle button
+- It will also allow workflow-designers to hide controls for any inputs that are set internally by the workflow. (Ex: x_by & y_by for a UMAP)
+
+Input structure:
+  'data-options': dictionary of options for '*_by' (ex: 'x_by', 'color_by') inputs where keys are the primary options and None vs [] values indicates what secondary options there may be if there are indeed secondary options (such as for genes!)
+  'hidden': list or null. The names of scatter_plotly inputs which should not be shown in the current widget render.  Perhaps these are hard-set by the workflow and a user has no choice.
+
+Output Structure:
+  dictionary of scatter_plotly inputs-name (key) + value pairs.
+*/
+
+const ScatterPlotly: InputBackendComponent = ({
+  input,
+  onChange
+}) => {
+  const {data, name} = input;
+
+  const options: {[k: string]: string} = useMemo(() => {
+    if (data == null) return {};
+    
+    let given = Object.values(data['data_options']);
+    let cols = Object.keys(data['data_frame'])
+
+    if (given == null && cols != null) {
```

Comment:
you could probably simplify this logic to `cols || given` which produces the logic `if cols is not null, return that, otherwise return given`.